src/systemcmds/netman/netman.cpp

```diff
@@ -0,0 +1,463 @@
+		router.u  =   ipcfg.router;
+		dnsaddr.u =   ipcfg.dnsaddr;
+		return *this;
+	}
+
+
+	int read(const char *netdev)
+	{
+		struct ipv4cfg_s ipcfg;
+		int rv = ipcfg_read(netdev, (FAR struct ipcfg_s *) &ipcfg, AF_INET);
+
+		if (rv == -EINVAL ||
+		    (rv == OK  && (ipcfg.proto > IPv4PROTO_FALLBACK || ipcfg.ipaddr == 0xffffffff))) {
+			// Build a default
+			ipcfg.ipaddr  = HTONL(DEFAULT_IP);
+			ipcfg.netmask = HTONL(DEFAULT_NETMASK);
+			ipcfg.router  = HTONL(DEFAULT_ROUTER);
+			ipcfg.dnsaddr = HTONL(DEFAULT_DNS);
+			ipcfg.proto   = DEFAULT_PROTO;
+			rv = ENOENT;
+		}
+
+		device.set_name(netdev);
+		*this = ipcfg;
+		return rv;
+	}
+
+	int write()
+	{
+		struct ipv4cfg_s ipcfg;
+		ipcfg.proto   = proto.e;
+		ipcfg.ipaddr  = ipaddr.u;
+		ipcfg.netmask = netmask.u;
+		ipcfg.router  = router.u;
+		ipcfg.dnsaddr = dnsaddr.u;
+		return ipcfg_write(device.name(), (FAR struct ipcfg_s *) &ipcfg, AF_INET);
+	}
+};
+
+int save(const char *path, const char *netdev)
+{
+
+	net_params config;
+	constexpr int lsz = 80;
+	char line[lsz + 1];
+	int len;
+
+	int rv = config.read(netdev);
+
+	int fd = open(path, O_CREAT | O_WRONLY | O_TRUNC, PX4_O_MODE_666);
+
+	if (fd < 0) {
+		PX4_ERR("Can not create file %s", path);
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n", config.device.keyword(), netdev);
+
+	if (len != write(fd, line, len)) {
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n",  config.proto.keyword(), config.proto.protocol());
+
+	if (len != write(fd, line, len)) {
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n", config.netmask.keyword(), config.netmask.to_str());
+
+	if (len != write(fd, line, len)) {
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n", config.ipaddr.keyword(), config.ipaddr.to_str());
+
+	if (len != write(fd, line, len)) {
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n", config.router.keyword(), config.router.to_str());
+
+	if (len != write(fd, line, len)) {
+		goto errout;
+	}
+
+	len = snprintf(line,  lsz, "%s%s\n", config.dnsaddr.keyword(), config.dnsaddr.to_str());
+
+	if (len != write(fd, line, len)) {
+		rv = -errno;
+
+	} else {
+		close(fd);
+		return rv;
+	}
+
+errout: {
+		rv = -errno;
+
+		if (fd >= 0) {
+			close(fd);
+		}
+
+		return rv;
+	}
+}
+
+int update(const char *path, const char *netdev)
+{
+	net_params config;
+	struct stat sb;
+	FAR char *lines = nullptr;
+	int fd = -1;
+	int rv = OK;
+
+	// First do we have a binary config stored?
+
+	rv = config.read(netdev);
+
+	if (rv == ENOENT) {
+		goto write_reboot;
+	}
+
+	// Is there a config file update?
+
+	if (stat(path, &sb) < 0) {
+		return 0;
+	}
+
+	lines = (char *) malloc(sb.st_size);
+
+	if (!lines) {
+		return -errno;
+	}
+
+	fd = open(path, O_RDONLY);
+
+	if (fd < 0) {
+		rv = -errno;
+		goto errout;
+	}
+
+	if (read(fd, lines, sb.st_size) != sb.st_size) {
+		rv = -errno;
+		goto errout;
+	}
+
+	close(fd);
+	fd = -1;
+	unlink(path);
+
+	config.proto.parse(lines, &lines[sb.st_size - 1]);
```

Comment:
Can you ensure `lines` is null-terminated? Since you pass it to `strstr`, which does not consider `end`.